rpcs3/Emu/VFS.cpp

```diff
@@ -33,7 +33,7 @@ struct vfs_manager
 	vfs_directory root{};
 };
 
-bool vfs::mount(std::string_view vpath, std::string_view path, bool create_dir)
+bool vfs::mount(std::string_view vpath, std::string_view path, bool create_dir, bool to_file, u64 file_size)
```

Comment:
This belongs to sys_fs_mount I'm sure. vfs::mount has enough bloat. VFS is already agnostic to whether the path is a directory or a file.

Reply:
Let me check what I can do later.

Reply:
@Nekotekina I think the parameter `bool to_file` is still needed for VFS::Mount(); otherwise, VFS::Mount() will append a '/' to any given `path`, breaking the path to a file.

Reply:
Why would you mount a file?

Reply:
> Why would you mount a file?

See the PR description, thank you : )

Reply:
@elad335 why would it append / to any given path? Savestates PR adds it.

Reply:
iirc resolve_path removes them so this is a fixup, can not append if didn't exust in the first place but it needs testing that everything works.

Reply:
I think this PR's implementation of VFS::Mount() is still acceptable though.
We could further clear up the VFS stuff after @elad335's concern regarding the VFS path is clarified.

Reply:
Cut one parameter.

Reply:
No, it's too ugly. I merged the fix.

Reply:
Done. I've rewritten the code.